src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -192,200 +193,199 @@ QRegularExpression AutoDownloadRule::cachedRegex(const QString &expression, bool
+                        return true;
                 }
             }
         }
-
-        return false;
+        else { // Single number
+            const QString expStr {QString("\\b(?:s0?%1[ -_\\.]?e0?%2|%1x0?%2)(?:\\D|\\b)").arg(season, episode)};
+            if (cachedRegex(expStr).match(articleTitle).hasMatch())
+                return true;
+        }
     }
 
-    if (useSmartFilter()) {
-        // now see if this episode has been downloaded before
-        const QString episodeStr = computeEpisodeName(articleTitle);
+    return false;
+}
+
+bool AutoDownloadRule::matchesSmartEpisodeFilter(const QString& articleTitle) const
+{
+    if (!useSmartFilter())
+        return true;
+
+    const QString episodeStr = computeEpisodeName(articleTitle);
+    if (episodeStr.isEmpty())
+        return true;
 
-        if (!episodeStr.isEmpty()) {
-            bool previouslyMatched = m_dataPtr->previouslyMatchedEpisodes.contains(episodeStr);
-            bool isRepack = articleTitle.contains("REPACK", Qt::CaseInsensitive) || articleTitle.contains("PROPER", Qt::CaseInsensitive);
-            if (previouslyMatched && !isRepack)
-                return false;
+    // See if this episode has been downloaded before
+    const bool previouslyMatched = m_dataPtr->previouslyMatchedEpisodes.contains(episodeStr);
+    const bool isRepack = articleTitle.contains("REPACK", Qt::CaseInsensitive) || articleTitle.contains("PROPER", Qt::CaseInsensitive);
+    if (previouslyMatched && !isRepack)
+        return false;
 
-            m_dataPtr->lastComputedEpisode = episodeStr;
-        }
+    m_dataPtr->lastComputedEpisode = episodeStr;
+    return true;
+}
+
+bool AutoDownloadRule::matches(const QVariantHash &articleData) const
+{
+    const QDateTime articleDate {articleData[Article::KeyDate].toDateTime()};
+    if (ignoreDays() > 0) {
+        if (lastMatch().isValid() && (articleDate < lastMatch().addDays(ignoreDays())))
+            return false;
     }
 
-//    qDebug() << "Matched article:" << articleTitle;
+    const QString articleTitle {articleData[Article::KeyTitle].toString()};
+    if (!matchesMustContainExpression(articleTitle))
+        return false;
+    if (!matchesMustNotContainExpression(articleTitle))
+        return false;
+    if (!matchesEpisodeFilterExpression(articleTitle))
+        return false;
+    if (!matchesSmartEpisodeFilter(articleTitle))
+        return false;
+
     return true;
 }
 
+bool AutoDownloadRule::accepts(const QVariantHash &articleData)
+{
+    if (!matches(articleData))
+        return false;
+
+    setLastMatch(articleData[Article::KeyDate].toDateTime());
+
+    if (!m_dataPtr->lastComputedEpisode.isEmpty()) {
+        // TODO: probably need to add a marker for PROPER/REPACK to avoid duplicate downloads
+        m_dataPtr->previouslyMatchedEpisodes.append(m_dataPtr->lastComputedEpisode);
+        m_dataPtr->lastComputedEpisode.clear();
+    }
```

Comment:
@glassez 
https://travis-ci.org/qbittorrent/qBittorrent/jobs/381660516#L1198

Reply:
Oops... shouldn't there be a compilation error in this case?
Will be fixed asap.

Reply:
Fixed in #8966.